res/controllers/Pioneer-DDJ-SR2-scripts.js

```diff
@@ -0,0 +1,1383 @@
+    this.cuepoint = 1;
+    this.range = PitchPlayRange.MID;
+
+    this.PerformancePad = function(n) {
+        this.midi = [0x97 + offset, 0x70 + n];
+        this.number = n + 1;
+        this.on = this.color + DDJSR2.PadColor.DIM_MODIFIER;
+        this.colorMapper = DDJSR2.PadColorMap;
+        this.colorKey = `hotcue_${  this.number  }_color`;
+        components.Button.call(this);
+    };
+    this.PerformancePad.prototype = new components.Button({
+        group: deck.currentDeck,
+        mode: this,
+        outConnect: false,
+        off: DDJSR2.PadColor.OFF,
+        unshift: function() {
+            this.outKey = "pitch_adjust";
+            this.output = function(_value, _group, _control) {
+                let color = this.mode.color + DDJSR2.PadColor.DIM_MODIFIER;
+                if ((this.mode.range === PitchPlayRange.UP && this.number === 5) ||
+                    (this.mode.range === PitchPlayRange.MID && this.number === 1) ||
+                    (this.mode.range === PitchPlayRange.DOWN && this.number === 4)) {
+                    color = DDJSR2.PadColor.WHITE;
+                }
+                this.send(color);
+            };
+            this.input = function(channel, control, value, _status, _group) {
+                if (value > 0) {
+                    let pitchAdjust;
+                    switch (this.mode.range) {
+                    case PitchPlayRange.UP:
+                        pitchAdjust = this.number + ((this.number <= 4) ? 4 : -5);
+                        break;
+                    case PitchPlayRange.MID:
+                        pitchAdjust = this.number - ((this.number <= 4) ? 1 : 9);
+                        break;
+                    case PitchPlayRange.DOWN:
+                        pitchAdjust = this.number - ((this.number <= 4) ? 4 : 12);
+                    }
+                    engine.setValue(this.group, "pitch_adjust", pitchAdjust);
+                    engine.setValue(this.group, `hotcue_${  this.mode.cuepoint  }_activate`, value);
+                }
+            };
+            this.connect = function() {
+                components.Button.prototype.connect.call(this); // call parent connect
+
+                if (this.connections[1] !== undefined) {
+                    // Necessary, since trigger() apparently also triggers disconnected connections
+                    this.connections.pop();
+                }
+            };
+            if (this.connections[0] !== undefined) {
+                this.disconnect();
+                this.connect();
+                this.trigger();
+            }
+        }
+    });
+    this.ShiftedPerformancePad = function(n) {
+        this.midi = [0x97 + offset, 0x78 + n];
+        this.number = n + 1;
+        this.on = this.color + DDJSR2.PadColor.DIM_MODIFIER;
+        this.colorMapper = DDJSR2.PadColorMap;
+        this.colorKey = `hotcue_${  this.number  }_color`;
+        components.Button.call(this);
+    };
```

Comment:
My question was unrelated to pitch play (I just filed the PR at a suboptimal place). If you do the procedure I outlined just on the plain `HotcueMode` pads, does the pad stay pressed? (you can test that by doing it when the track is paused, if the tracks continues to play after you released the button, it didn't register the release).

Reply:
If you hold down hotcue, to play a hotcue on a track that's not already playing (so playing is reliant on the hotcue being held) press shift, it will still play, but if you let go of the hotcue button, it will stop.

Reply:
Okay thats good to know. Thanks for the confirmation.